### PR TITLE
KT-47068: Move user groups data to YML

### PR DESCRIPTION
**Pull request by krzema12**
`data/user-groups.yml` is now the source of truth about user groups.

Markdown -> YAML conversion was done by a script pasted in the
comment of the below issue + one manual adjustment when a single
Markdown bullet point spanned two lines in raw text.

# Issue

https://youtrack.jetbrains.com/issue/KT-47068

# Tests

Ran locally and tested that:
* header anchors work
* the JSON file is not loaded when opening another community-related page
* the server doesn't start if there's a syntax error in the YAML file

Also, compared pixel-wise with the current version in prod. The only significant visible difference is around some "India" word which had an extra space in the Markdown version, and doesn't have it when the page is generated with JS. Other visible minor differences are regarding font antialiasing.

image

### CODE REVIEW

**Review by nikpachoo — CHANGES_REQUESTED** [review]
Thanks for the work you've done!
Migration to the yaml file is correct, looking good.

The only thing is, I would like to keep the rendering on the server-side. For this, you need:
- create a Jinja2 template in the templates / pages directory, with  'groups.html' extends, for example:
  ```twig
  {% set restyled="v1" %}
  {% extends 'groups.html' %}
  
  
  {% block page_content %}
      place content from "pages/user-groups/user-group-list.md" here
  {% endblock %}
  ```
- add a page route to the "kotlin-website.py",  for example:
   ```py
   @app.route('/user-groups/user-group-list.html')
   def user_group_list():
     return render_template('pages/user-groups/user-group-list.html')
   ```
- pass the YML-data to the created route and create a markup 
- delete the "pages/user-groups/user-group-list.md" page as obsolete.

At the same time, I would like to keep the route "/data/user-groups.json" for getting the YML data for the User Group map. 

@krzema12 what do you think? If you need help, I can help you with this.

**Comment by krzema12** [comment]
Thanks @nikpachoo, makes perfect sense! I wasn't aware that the server-side templates are already used somewhere. Your suggestion is conceptually clear to me, I'll give it a try one day and will let you know in case of any difficulties.

**Comment by krzema12** [comment]
@nikpachoo I applied your suggestion and rebased on newest master so that the data in the YML is up-to-date.

One unrelated thing: something's wrong on master - running it locally and opening the website fails. To unblock myself, I had to comment out this line:

https://github.com/JetBrains/kotlin-web-site/blob/4a93ddc88a8867b56d3d3615db7f0d5ced79e1a4/templates/base.html#L22

File: kotlin-website.py
Comment (krzema12) [inline kotlin-website.py:246]: I don't know how to perform such nested summing in Jinja2, so I decided to pass another template argument. If you know a way to move it to the template, I'm happy to do follow your hint.

**Comment by nikpachoo** [comment]
> One unrelated thing: something's wrong on master - running it locally and opening the website fails. To unblock myself, I had to comment out this line:
> 
> https://github.com/JetBrains/kotlin-web-site/blob/4a93ddc88a8867b56d3d3615db7f0d5ced79e1a4/templates/base.html#L22

Oh, that is true, you need to run "yarn install" on the "scripts/react-renderer" directory to fix it.
Sorry for the inconvenience, we are going to fix it next week.

File: kotlin-website.py
Comment (nikpachoo) [reply kotlin-website.py:246]: I think you have chosen the right way. It looks clear.

**Review by nikpachoo — APPROVED** [review]

**Comment by krzema12** [comment]
It's really easy to get conflicts for `user-group-lost.md` :D Hoping that you can resolve them just after giving approval and before merging, otherwise the conflict may appear again and again.

**Comment by zoobestik** [comment]
@krzema12 great job! 👍  Could you resolve conflict with [Alina commit](https://github.com/JetBrains/kotlin-web-site/commit/1a64549494f2870e819c77d951935ff67db4010d#diff-971f1b63559f9eca6c8e7719d236a95d17135b054c13495e9d4f168fa00a70dd)?

 * [Chennai Kotlin User Group](https://discord.gg/UYHG85Pgu9), India
 * [Kano Kotlin User Group](https://web.facebook.com/groups/193660894904095/about), Nigeria
 * [Minna Kotlin User Group](https://web.facebook.com/groups/969877396554431/about), Nigeria

**Comment by krzema12** [comment]
@zoobestik done :)

**Review by zoobestik — APPROVED** [review]

**Pull request merged**
